src/utils.rs

```diff
@@ -0,0 +1,118 @@
+// This file is part of the uutils diffutils package.
+//
+// For the full copyright and license information, please view the LICENSE-*
+// files that was distributed with this source code.
+
+use std::io::Write;
+
+use unicode_width::UnicodeWidthStr;
+
+/// Replace tabs by spaces in the input line.
+/// This assumes that line does not contain any line breaks (if it does, the result is undefined).
+pub fn do_expand_tabs(line: &[u8], tabsize: usize) -> Vec<u8> {
+    // This matches GNU's implementation which handles correctly multi-bytes characters since
+    // https://git.savannah.gnu.org/cgit/diffutils.git/commit/?id=924f297.
```

Comment:
this is gpl code, please don't reference it

Reply:
Understood. Note that the implementation didn't draw inspiration from GNU's implementation, it merely respected the claim that this feature is correctly handled (since recently). But I can see where this might be considered dubious, so I removed the reference.